### PR TITLE
ddl: update schema version after the execution

### PR DESCRIPTION
**Pull request by xhebox**
### What problem does this PR solve?

Problem Summary: Some of the DDL jobs will invoke `updateSchemaVersion` before actually executing the job. But I want to pass the result of execution to `updateSchemaVersion`: the result of `getIDs()` in `onDropSchema` could be used to drop placement rules cache.

Moreover, `SchemaDiff` in `updateSchemaVersion` should be set after the execution. That is what other DDL jobs are doing.

**UPDATE**: Looks like it is needed to do so. `onDropSchema` has three schema states, thus will invoke `updateSchemaVersion->SetSchemaDiff->ApplyDiff` three times. That is if I want to drop rule cache precisely at the last schema state, I need extra info to `updateSchemaVersion`.

Or I have to drop cache/notify PD even if there is no rule bundle for this schema. That will increase undesire network costs.

Or I must drop cache at the first schema state of `onDropSchema`, which is incorrect, IMO.

I don't think it is correct to call `ApplyDiff->b.applyDropSchema` three times. DDL job queue should be updated to demonstrate its schema state, but the metadata should only be dropped at the last schema state. This PR will give a chance for further improvements.

### Check List 

Tests 

- Unit test
- Integration test

### Release note 

- No release note

### CODE REVIEW

File: ddl/schema.go [hunk 1]
```diff
@@ -54,13 +49,17 @@ func onCreateSchema(d *ddlCtx, t *meta.Meta, job *model.Job) (ver int64, _ error
 		if err != nil {
 			return ver, errors.Trace(err)
 		}
+		ver, err = updateSchemaVersion(t, job)
+		if err != nil {
+			return ver, errors.Trace(err)
+		}
```
Comment (AilinKid) [inline ddl/schema.go:55]: updateSchemaVersion and the check logic is in the same transaction t, they could be done atomically. 
That's means the updateScheamVersion + DropSchema + GetIDs() are atomic, either all succeed or all fail.

Comment (xhebox) [reply ddl/schema.go:55]: I did not mean it is not atomic. But I want to call `result = getIDs()` first before I set `job.CtxVars = []interface{}{result};updateSchemaVersion(...);`.

Comment (djshow832) [reply ddl/schema.go:55]: Why do you need the result of `getIDs`? You can also get it by `ListTables` after execution, in `ApplyDiff`.

Comment (xhebox) [reply ddl/schema.go:55]: I need to invoke `infosync.PutRuleBundles` to drop rules anyway. I don't want to get it twice: once in the DDL jobs, and another in `ApplyDiff`.

And, of course, I can run `ListTables`. It is indeed better that it did not use the confusing `AffectedOpts`.

Then, the only point for this PR is consistent behavior/usage of `updateSchemaVersion`. If you think that is unnecessary, I will close the PR.

**Pull request closed without merging**